conans/client/tools/pkg_config.py

```diff
@@ -98,3 +98,14 @@ def variables(self):
             for name in variable_names:
                 self._variables[name] = self._parse_output('variable=%s' % name)
         return self._variables
+
+
+def fill_cppinfo_from_pc_file(conanfile, library, pkg_config_executable='pkg-config', static=False, msvc_syntax=False,
+                              variables=None, print_errors=True):
+    pkg_config = PkgConfig(library, pkg_config_executable, static, msvc_syntax, variables, print_errors)
+    libs = [lib[2:] for lib in pkg_config.libs_only_l]  # cut -l prefix
+    lib_paths = [lib[2:] for lib in pkg_config.libs_only_L]  # cut -L prefix
+    conanfile.cpp_info.libs.extend(libs)
+    conanfile.cpp_info.libdirs.extend(lib_paths)
+    conanfile.cpp_info.sharedlinkflags.extend(pkg_config.libs_only_other)
+    conanfile.cpp_info.exelinkflags.extend(pkg_config.libs_only_other)
```

Comment:
Several questions here:
 * Why extending instead of substituting?
 * Why not return a `CppInfo` object and assign it to the conanfile member? (it would be easier to make more tests)
 * What about other options considered in `cpp_info` like `cflags` or `libs-only-other`?
 * Does it return the values with the substitutions (`${prefix}`,...) already done?

Reply:
1. in this case, it will override already filled cpp_info, right? also, two sequential calls to `fill_cppinfo_from_pc_file` will result in libraries only from the last one
2. I don't think I usually need one more `CppInfo` object instance in my recipe's conan file, as I already have `self.cpp_info`
3. as it's designed to add only link-time dependencies, to link with already compiled library, there is not much sense of adding cflags
4. yes, already prefixed paths

Reply:
Ok, now I understand this better.